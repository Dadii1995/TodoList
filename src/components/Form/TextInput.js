import React from 'react'
import { FormGroup, Label, Input, FormFeedback } from 'reactstrap'

const TextInput = ({ field: { name, ...field }, form: { touched, errors }, ...props }) => (
  <FormGroup>
    {props.label && <Label>{props.label}</Label>}
    <Input
      autoComplete="off"
      innerRef={props.innerRef}
      type="text"
      {...field}
      aria-label={props.label}
      data-cy={props.dataCy}
      name={name}
      placeholder={props.placeholder}
    />
    {errors[name] && <FormFeedback>{errors[name]}</FormFeedback>}
  </FormGroup>
)

export default TextInput
