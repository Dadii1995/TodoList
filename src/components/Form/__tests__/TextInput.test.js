import renderer from 'react-test-renderer'
import React from 'react'
import { Field, Formik } from 'formik'
import TextInput from '../TextInput'

const getWrapper = initialState => {
  const wrapper = renderer.create(
    <Formik initialValues={initialState.initialValues}>
      <Field
        component={TextInput}
        label={initialState.label}
        name={initialState.name}
        placeholder={initialState.placeholder}
      />
    </Formik>,
  )
  const root = wrapper.root
  return { root, wrapper }
}
describe('<TextInput/>', () => {
  test('render text input with label', () => {
    const initialState = {
      name: 'textInput',
      label: 'Content:',
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
  test('render text input with placeholder', () => {
    const initialState = {
      name: 'textInput',
      placeholder: 'Content:',
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
  test('render text input with label and placeholder', () => {
    const initialState = {
      name: 'textInput',
      placeholder: 'Content:',
      label: 'Label',
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
  test('render text input without label and placeholder', () => {
    const initialState = {
      name: 'textInput',
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
  test('render textarea with label and initial value', () => {
    const initialState = {
      name: 'textInput',
      label: 'Label',
      initialValues: {
        textArea: 'Initial content',
      },
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
})
