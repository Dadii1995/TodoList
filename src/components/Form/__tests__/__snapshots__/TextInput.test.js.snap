// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> render text input with label 1`] = `
<div
  className="form-group"
>
  <label
    className=""
  >
    Content:
  </label>
  <input
    aria-label="Content:"
    autoComplete="off"
    className="form-control"
    name="textInput"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
</div>
`;

exports[`<TextInput/> render text input with label and placeholder 1`] = `
<div
  className="form-group"
>
  <label
    className=""
  >
    Label
  </label>
  <input
    aria-label="Label"
    autoComplete="off"
    className="form-control"
    name="textInput"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Content:"
    type="text"
  />
</div>
`;

exports[`<TextInput/> render text input with placeholder 1`] = `
<div
  className="form-group"
>
  <input
    autoComplete="off"
    className="form-control"
    name="textInput"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Content:"
    type="text"
  />
</div>
`;

exports[`<TextInput/> render text input without label and placeholder 1`] = `
<div
  className="form-group"
>
  <input
    autoComplete="off"
    className="form-control"
    name="textInput"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
</div>
`;

exports[`<TextInput/> render textarea with label and initial value 1`] = `
<div
  className="form-group"
>
  <label
    className=""
  >
    Label
  </label>
  <input
    aria-label="Label"
    autoComplete="off"
    className="form-control"
    name="textInput"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
</div>
`;
